pkg/test/framework/components/echo/config.go

```diff
@@ -72,6 +72,18 @@ type Config struct {
 	// ReadinessTimeout specifies the timeout that we wait the application to
 	// become ready.
 	ReadinessTimeout time.Duration
+
+	// Workloads contains the list of workloads config belonging to this echo service instance.
+	Workloads []WorkloadConfig
+}
+
+// WorkloadConfig is the config for a group of workloads (e.g. Kubernetes deployment).
+type WorkloadConfig struct {
+	Version string
+	// Name is the name of the workload.
+	Name        string
+	Annotations Annotations
```

Comment:
Rather than having Annotations at the top level and under workload, why not just move everything into the Workloads array? So if there is only 1 element, it would behave as before.

This also implies that we don't merge annotations, which I think is a good idea anyway.

Reply:
just to confirm, you want to then change all caller's construction of `echo.Config` to specify config.Workloads.Annotations, and not doing conversions internally at component library itself?

that'll be fine, but might turn into a bigger PR... i can see the feasiblity

Reply:
addressed.